index-configuration-tool/main.py

```diff
@@ -0,0 +1,135 @@
+import sys
+import requests
+import logstash_conf_parser as parser
+from jsondiff import diff
+
+# Constants
+SUPPORTED_ENDPOINTS = ["opensearch", "elasticsearch"]
+HOSTS_KEY = "hosts"
+USER_KEY = "user"
+PWD_KEY = "password"
+SETTINGS_KEY = "settings"
+MAPPINGS_KEY = "mappings"
+INDEX_KEY = "index"
+INTERNAL_SETTINGS_KEYS = ["creation_date", "uuid", "provided_name", "version", "store"]
+ALL_INDICES_ENDPOINT = "*"
+
+
+# Utility method to make a comma-separated string from a set
+def string_from_set(s: set) -> str:
+    if s:
+        return ", ".join(s)
+    else:
+        return "[]"
```

Comment:
Yes, the example above is correct. I'd like to communicate to users that a set is explicitly empty rather than printing an empty string, hence the `[]` return value. Since you noted that this is unusual, is there a better way to achieve this?

I'll update the type annotation to `set[str]` as you've suggested.

Reply:
Mm, okay, I see what you mean. I'm okay with the behavior -- it looks very asymmetric on its own, but the way it works in context seems reasonable.